tests/test_plugin_binance_com.py

```diff
@@ -0,0 +1,652 @@
+                "type": "limit",  # order type, 'market', 'limit' or undefined/None/null
+                "side": "sell",  # direction of the trade, 'buy' or 'sell'
+                "takerOrMaker": "taker",  # string, 'taker' or 'maker'
+                "price": 23000.01,  # float price in quote currency
+                "amount": 1,  # amount of base currency
+                "cost": 23000.01,  # total cost, `price * amount`,
+                "fee": {  # provided by exchange or calculated by ccxt
+                    "cost": 40,  # float
+                    "currency": "GBP",  # usually base currency for buys, quote currency for sells
+                    "rate": 0.002,  # the fee rate (if available)
+                },
+            },
+        ]
+
+        # CCXT abstracts dust trades into regular trades, so no testing is necessary
+        mocker.patch.object(plugin.client, "fetch_my_dust_trades").return_value = []
+        mocker.patch.object(plugin, "_process_deposits").return_value = None
+        mocker.patch.object(plugin, "_process_gains").return_value = None
+        mocker.patch.object(plugin, "_process_withdrawals").return_value = None
+
+        result = plugin.load()
+
+        # One Sell of quote asset (using BNB) +
+        # One Buy of base asset (using BNB) +
+        # One payment for fees in BNB
+        # One Sell of quote asset (for Buy order) +
+        # One Buy of base asset (for Buy order) +
+        # One Sell of quote asset (for Sell order) +
+        # One Buy of base asset (for Sell order) +
+        # One Buy of base asset (for fiat buy) +
+        # One Sell of quote asset (for fiat buy) +
+        # One Sell of quote asset (for fiat sell) +
+        # One Buy of base asset (for fiat buy) = 11
+        assert len(result) == 11
+
+        bnb_sell_transaction: OutTransaction = result[5]  # type: ignore
+        bnb_buy_transaction: InTransaction = result[0]  # type: ignore
+        bnb_fee_transaction: OutTransaction = result[6]  # type: ignore
+        buy_conversion_out: OutTransaction = result[7]  # type: ignore
+        buy_conversion_in: InTransaction = result[1]  # type: ignore
+        sell_conversion_order_out: OutTransaction = result[8]  # type: ignore
+        sell_conversion_order_in: InTransaction = result[2]  # type: ignore
+        buy_fiat_order_out: OutTransaction = result[9]  # type: ignore
+        buy_fiat_order_in: InTransaction = result[3]  # type: ignore
+        sell_fiat_order_out: OutTransaction = result[10]  # type: ignore
+        sell_fiat_order_in: InTransaction = result[4]  # type: ignore
+
+        # Buy with BNB as fee payment
+        assert bnb_sell_transaction.asset == "BTC"
+        assert int(parser.parse(bnb_sell_transaction.timestamp).timestamp()) * 1000 == 1502962946000
+        assert bnb_sell_transaction.transaction_type == Keyword.SELL.value
+        assert bnb_sell_transaction.spot_price == Keyword.UNKNOWN.value

```

Comment:
Based on the data structure at line 198 it does look like we cannot compute `spot_price` (because the fiat amount is missing): this is a bit strange. Does Binance not provide fiat amount for this conversion (perhaps it's available in some other data structure)? If not I guess we can rely on the transaction resolver to fill `spot_price`, but it's surprising because it seems that a basic piece of info is missing (Coinbase does provide it).

Reply:
Yeah, exchanges based in only one country like Coinbase US, would have the fiat spot price since there is only one fiat in use in a country, generally speaking. However, any international exchange would not be able to do this since there is no 'base' currency. Binance.com has trading pairs in the British Pound (GBP), the Ruble (RUB) and even the Ukrainian Hryvnia (UAH). 

Technically, Binance.com doesn't have any USD pairs, so any total would be an estimate. So, I think they just want to be safe and leave it up to the user to generate the estimates.

Reply:
Ok, got it. I think the plan we came up with (pair converters, country plugins, etc.) should be robust enough to take care of this: we'll just let the transaction resolver do the heavy lifting.